### PR TITLE
Move semantics stuff from rendering to a top-level semantics/ library

### PR DESCRIPTION
**Pull request by amirh**
This is based off https://github.com/flutter/flutter/pull/12789

### CODE REVIEW

File: packages/flutter/lib/semantics.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// The Flutter rendering tree.
```
Comment (goderbauer) [inline packages/flutter/lib/semantics.dart]: This doc comment probably needs to be updated to talk about semantics instead of rendering?

Comment (amirh) [reply packages/flutter/lib/semantics.dart]: Good catch, I wrote a new comment, see if it makes sense

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 2]
```diff
@@ -197,7 +198,7 @@ class SemanticsData extends Diagnosticable {
   }
 
   @override
-  int get hashCode => hashValues(flags, actions, label, textDirection, rect, tags, transform);
+  int get hashCode => ui.hashValues(flags, actions, label, textDirection, rect, tags, transform);
```
Comment (Hixie) [inline packages/flutter/lib/src/semantics/semantics.dart]: you can show hashValues in the import

**Comment by Hixie** [comment]
LGTM

**Pull request merged**